### PR TITLE
Keyboard friendly navigation in Coordinape

### LINKED ISSUE #1432: Keyboard friendly allocations notes in the give beta build
### What's happening:
Currently navigating on the allocation screen is not keyboard friendly.
### What should be happening:
When you open the allocation screen your cursor should start on the allocation amount, and tab should take you to the notes, and tab again should take you to the next user

The tab flow should be:

1. Allocation amount
2. Note
3. Next Member button
4. Repeat

### PR DESCRIPTION
**Pull request by wingfeatherwave**
resolves #1432

* Lots of navigation updates so you can use `tab` to get around the site.  
* in some components, like our popover menus, you can use `return` to open them up, `escape` to close
* in allocations, when you closing a drawer you will keep your tab focus

### CODE REVIEW

**Comment by wingfeatherwave** [comment]
why did my commit history get duplicated when I rebased main locally?

**Comment by reeserj** [comment]
@wingfeatherwave When I create a new contribution from the Keyboard and tab to the close button and hit enter tabbing again i have to go back through the header to get to a new contribution. 

Is is possible to have it pop back out on the add contribution button from there?

File: src/components/MainLayout/OverviewMenu.tsx [hunk 1]
```diff
@@ -182,7 +196,14 @@ export const OverviewMenu = ({
                     <Link
                       key={circle.id}
                       type="menu"
-                      onClick={() => closeAndGo(paths.history(circle.id))}
+                      // onClick={() => closeAndGo(paths.history(circle.id))}
```
Comment (crabsinger) [inline src/components/MainLayout/OverviewMenu.tsx:199]: comment remove

File: src/pages/GivePage/EpochStatementDrawer.tsx [hunk 2]
```diff
@@ -200,6 +200,8 @@ export const EpochStatementDrawer = ({
           value={statement}
           onChange={e => statementChanged(e.target.value)}
           placeholder="Summarize your Contributions"
+          // eslint-disable-next-line jsx-a11y/no-autofocus
```
Comment (crabsinger) [inline src/pages/GivePage/EpochStatementDrawer.tsx:203]: ? what's this for?

Comment (crabsinger) [reply src/pages/GivePage/EpochStatementDrawer.tsx:203]: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-autofocus.md

Comment (crabsinger) [reply src/pages/GivePage/EpochStatementDrawer.tsx:203]: hmm ok

File: src/pages/GivePage/MyGiveRow.tsx [hunk 3]
```diff
@@ -28,11 +31,34 @@ export const MyGiveRow = ({
...
   optOutOpen: boolean;
   statementCompelete: boolean;
+  selected: boolean;
 }) => {
+  const newRef = useRef<HTMLButtonElement>(null);
+  const [lastSelected, setLastSelected] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (selected) {
+      // eslint-disable-next-line no-console
```
Comment (crabsinger) [inline src/pages/GivePage/MyGiveRow.tsx:41]: remove consoles...

File: src/pages/GivePage/index.tsx [hunk 4]
```diff
@@ -713,6 +697,10 @@ const AllocateContents = ({
         contributionCount={
           myMember?.contributions_aggregate?.aggregate?.count ?? 0
         }
+        // selected={true}
```
Comment (crabsinger) [inline src/pages/GivePage/index.tsx:700]: remove

File: src/pages/GivePage/index.tsx [hunk 5]
```diff
@@ -818,7 +744,8 @@ const AllocateContents = ({
           paddingTop: 0,
           overflowY: 'scroll',
         }}
-        onClose={() => {
+        onOpenChange={() => {
+          // giveRowFocus()
```
Comment (crabsinger) [inline src/pages/GivePage/index.tsx:748]: remove

File: src/ui/Modal.tsx [hunk 6]
```diff
@@ -61,7 +67,6 @@ const Title = styled(Dialog.Title, {
 type ModalProps = {
   children: ReactNode;
   title?: string;
-  onClose: () => void;
```
Comment (crabsinger) [inline src/ui/Modal.tsx:64]: should we make `onOpenChange` required? Or is it only needed to overwrite normal functionality?

File: src/pages/GivePage/EpochStatementDrawer.tsx (hunk 2 above)
Comment (wingfeatherwave) [reply src/pages/GivePage/EpochStatementDrawer.tsx:203]: This is to ignore the accessibility warning about hijacking focus.  in this case we are returning the user's focus to the drawer trigger they clicked, which is the ideal a11y behavior

File: src/ui/Modal.tsx (hunk 6 above)
Comment (wingfeatherwave) [reply src/ui/Modal.tsx:64]: hrm.   maybe we should.  but technically if we're just using a radix trigger to manage the open state, then there is no need for open or onOpenChange

File: src/components/MyAvatarMenu/MyAvatarMenu.tsx [hunk 7]
```diff
@@ -44,9 +44,18 @@ export const MyAvatarMenu = () => {
       <Hidden smDown>
         <Popover open={mouseEnterPopover}>
           <PopoverTrigger
+            tabIndex={-1}
             css={{ outline: 'none' }}
-            asChild
             ref={triggerRef}
+            onKeyDown={e => {
+              if (e.key === 'Enter') {
```
Comment (CryptoGraffe) [inline src/components/MyAvatarMenu/MyAvatarMenu.tsx:51]: This condition is different than the one in OverviewMenu. maybe a lil helper function that is like:
```
const onEnter = (f: func(e: uhhSomething): void) => {
 if (e.key === 'Enter' || e.key === ' ') {
   f(e);
 }
}
```

File: src/pages/GivePage/index.tsx [hunk 8]
```diff
@@ -748,69 +735,7 @@ const AllocateContents = ({
...
-                  left: 0,
-                  zIndex: 2,
-                }}
-              ></Box>
-            </Box>
-          </Flex>
-        </Box>
-      )}
-
+      {}
```
Comment (CryptoGraffe) [inline src/pages/GivePage/index.tsx:738]: Whatsup with this change? This deletes the example of how to add a contributor, which was in the design

File: src/ui/Modal.tsx [hunk 9]
```diff
@@ -88,22 +92,12 @@ export const Modal = ({
       onOpenChange={onOpenChange}
     >
       <Dialog.Portal>
-        <Overlay onClick={onClose} />
```
Comment (CryptoGraffe) [inline src/ui/Modal.tsx:91]: I like overly onclick close, why go bye bye?

Comment (wingfeatherwave) [reply src/ui/Modal.tsx:91]: it still does.  just letting radix methods handle it via onOpenChange

File: src/components/MyAvatarMenu/MyAvatarMenu.tsx (hunk 7 above)
Comment (wingfeatherwave) [reply src/components/MyAvatarMenu/MyAvatarMenu.tsx:51]: good idea.  will add.

File: src/pages/GivePage/index.tsx (hunk 8 above)
Comment (wingfeatherwave) [reply src/pages/GivePage/index.tsx:738]: will replace.

**Review by CryptoGraffe — APPROVED** [review]
Approved with the caveat that we need a new ticket that fixes the clearing out the give allocator, wing and I discussed the details here

**Pull request merged**